tendermint/src/abci/types.rs

```diff
@@ -0,0 +1,304 @@
+//! ABCI-specific data types used in requests and responses.
+//!
+//! These types have changes from the core data structures to better accomodate
+//! ABCI applications.
+//!
+//! [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#data-types)
+
+use std::convert::{TryFrom, TryInto};
+
+use bytes::Bytes;
+use chrono::{DateTime, Utc};
+
+use crate::PublicKey;
+
+/// A validator address with voting power.
+///
+/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#validator)
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Validator {
+    /// The validator's address (the first 20 bytes of `SHA256(public_key)`).
+    pub address: [u8; 20],
+    /// The voting power of the validator.
+    pub power: i64,
+}
+
+/// A change to the validator set.
+///
+/// Used to inform Tendermint of changes to the validator set.
+///
+/// [ABCI documentation](https://docs.tendermint.com/master/spec/abci/abci.html#validatorupdate)
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ValidatorUpdate {
+    /// The validator's public key.
+    pub pub_key: PublicKey,
+    /// The validator's voting power.
+    pub power: i64,
```

Comment:
Could the `Power` struct in `tendermint/src/vote/power.rs` be useful here?

Reply:
Probably!